Source/WebKit/WebProcess/WebCoreSupport/WebPermissionController.cpp

```diff
@@ -86,11 +89,23 @@ void WebPermissionController::removeObserver(WebCore::PermissionObserver& observ
     m_observers.remove(observer);
 }
 
-void WebPermissionController::permissionChanged(const WebCore::ClientOrigin& origin, const WebCore::PermissionDescriptor& descriptor, WebCore::PermissionState state)
+void WebPermissionController::permissionChanged(WebCore::PermissionName permissionName, const WebCore::SecurityOriginData& topOrigin)
 {
     for (auto& observer : m_observers) {
-        if (observer.origin() == origin && observer.descriptor() == descriptor)
-            observer.stateChanged(state);
+        if (observer.origin().topOrigin == topOrigin && observer.descriptor().name == permissionName) {
+            auto& context = observer.context();
+            auto source = WebCore::Permissions::sourceFromContext(context);
+            // FIXME: Add support for workers.
+            ASSERT(source && *source == WebCore::PermissionQuerySource::Window);
+            auto* page = downcast<WebCore::Document>(context).page();
+
+            if (page) {
+                query(WebCore::ClientOrigin { observer.origin() }, WebCore::PermissionDescriptor { permissionName }, page, *source, [observer = WeakPtr { observer }](auto newState) {
```

Comment:
PermissionDescriptor is just an enum, no need to pass it as r-value reference. Please update query() to take a `PermissionDescriptor` instead of a `PermissionDescriptor&&`. Then you can drop the `WebCore::PermissionDescriptor { }` here.

Reply:
That's true. PermissionDescriptor is a struct that only contains a PermissionName, so there's no need to pass it in as an r-value reference. But I believe we can't drop the `WebCore::PermissionDescriptor { }`.